readthedocs/oauth/models.py

```diff
@@ -84,6 +87,78 @@ class GitHubAppInstallation(TimeStampedModel):
     class Meta(TimeStampedModel.Meta):
         verbose_name = _("GitHub app installation")
 
+    @cached_property
+    def service(self):
+        """Return the service for this installation."""
+        from readthedocs.oauth.services import GitHubAppService
+
+        return GitHubAppService(self)
+
+    def delete(self, *args, **kwargs):
+        """Override delete method to remove orphaned organizations."""
+        self.delete_repositories()
+        return super().delete(*args, **kwargs)
+
+    def delete_repositories(self, repository_ids: list[int] | None = None):
+        """
+        Delete repositories linked to this installation.
+        When an installation is deleted, we delete all its remote repositories
+        and relations, users will need to manually link the projects to each repository again.
+        We also remove organizations that don't have any repositories after removing the repositories.
+        :param repository_ids: List of repository ids (remote ID) to delete.
+         If None, all repositories will be considered for deletion.
+        """
+        if repository_ids is not None and not repository_ids:
```

Comment:
I'm a little confused what this is checking for... an empty list? It seems like it could be more explicit if so.